src/nrnmpi/nrnmpi_dynam.cpp

```diff
@@ -232,9 +232,16 @@ std::string nrnmpi_load(int is_python) {
         return name;
     };
     auto const nrn_mpi_library = mpi_path("nrnmpi_");
-    // TODO this will be wrong if CoreNEURON is installed externally
     corenrn_mpi_library = mpi_path("corenrnmpi_");
 
+    // This env variable is only needed in usage like neurodamus where
+    // `solve_core()` is directly called by MOD file and it doesn't have
+    // an easy way to know which MPI library to load.
+    // TODO: remove when BlueBrain/neurodamus/issues/XX is fixed.
+#if defined(HAVE_SETENV)
+    setenv("NRN_CORENRN_MPI_LIB", corenrn_mpi_library.c_str(), 0);
+#endif
+
```

Comment:
@olupton : as mentioned earlier, I am thinking of this "workaround" with minimal changes until neurodamus team fixes BlueBrain/neurodamus/issues/17.

There could be other solutions but for the one specific usage in neurodamus, I am thinking more/extra changes are not worth it (?).

Thoughts?